lib/util/http/github.ts

```diff
@@ -210,31 +210,26 @@ export class GithubHttp extends Http<GithubHttpOptions, GithubHttpOptions> {
       if (opts.paginate) {
         // Check if result is paginated
         const pageLimit = opts.pageLimit ?? 10;
-        const linkHeader =
-          result?.headers?.link &&
-          parseLinkHeader(result.headers.link as string);
+        const linkHeader = is.string(result?.headers?.link)
+          ? parseLinkHeader(result.headers.link)
+          : null;
         if (linkHeader?.next && linkHeader?.last) {
-          let lastPage = +linkHeader.last.page;
+          let lastPage = Number(linkHeader.last.page);
           // istanbul ignore else: needs a test
           if (!process.env.RENOVATE_PAGINATE_ALL && opts.paginate !== 'all') {
             lastPage = Math.min(pageLimit, lastPage);
           }
-          const pageNumbers = Array.from(
-            new Array(lastPage),
-            (x, i) => i + 1
-          ).slice(1);
-          const queue = pageNumbers.map(
-            (page) => (): Promise<HttpResponse<T>> => {
+          const queue = Array.from(new Array(lastPage), (_, i) => `${i + 1}`)
```

Comment:
That looks really ugly, we should replace it with a generator function like in test in a future PR.